cmd/admin/auth.go

```diff
@@ -102,8 +103,45 @@ func handlerAuthCheck(h http.Handler) http.Handler {
 				samlMiddleware.RequireAccount(h).ServeHTTP(w, r)
 			}
 		case settings.AuthHeaders:
-			// Access always granted
-			h.ServeHTTP(w, r)
+			username := r.Header.Get(headersConfig.TrustedPrefix + headersConfig.UserName)
+			groups := strings.Split(r.Header.Get(headersConfig.TrustedPrefix+headersConfig.Groups), ",")
+
+			if username == "" {
+				log.Printf("A username is required to use this system.")
+				w.WriteHeader(http.StatusBadRequest)
```

Comment:
In other authentication methods I have used a redirect instead of writing the header:
```
http.Redirect(w, r, forbiddenPath, http.StatusFound)
```
Do you think we should use `WriteHeader` instead?